packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js

```diff
@@ -34,122 +37,139 @@ function getActionHandler(func, item) {
  * @returns XML, jsx to display
  */
 
-function TreeViewItem({ id, item, depth = 0, onClick, onSelect }) {
-	const {
-		toggled = false,
-		selected,
-		hidden,
-		name,
-		children = [],
-		showCounter,
-		actions,
-		icon = 'talend-folder',
-		counter = children.length,
-	} = item;
-	const toggleIconLabel = toggled ? 'Collapse' : 'Expand';
-
-	function onSelectWrapper() {
-		return onSelect(item);
+class TreeViewItem extends React.Component {
+	static propTypes = {
+		id: PropTypes.string.isRequired,
+		item: PropTypes.shape({
+			name: PropTypes.string.isRequired,
+			toggled: PropTypes.bool,
+			selected: PropTypes.bool,
+			children: PropTypes.arrayOf(PropTypes.object),
+			icon: PropTypes.string,
+			actions: PropTypes.arrayOf(
+				PropTypes.shape({
+					action: PropTypes.func,
+					label: PropTypes.string,
+					icon: PropTypes.string,
+				}),
+			),
+			counter: PropTypes.number,
+			showCounter: PropTypes.bool,
+		}).isRequired,
+		onClick: PropTypes.func.isRequired,
+		onSelect: PropTypes.func.isRequired,
+		depth: PropTypes.number,
+	};
+
+	static defaultProps = {
+		id: 'tc-treeview-item',
+		depth: 0,
+	};
+
+	constructor(props) {
+		super(props);
+		this.onSelect = this.onSelect.bind(this);
+		this.onKeyDown = this.onKeyDown.bind(this);
+		this.renderTreeViewItem = this.renderTreeViewItem.bind(this);
 	}
 
-	function getTreeViewItem(child, i) {
+	onKeyDown(event) {
+		if (event.keyCode === keycode.codes.enter) {
+			this.onSelect();
+		}
+	}
+
+	onSelect() {
+		this.containerRef.focus();
+		this.props.onClick(this.props.item);
+		return this.props.onSelect(this.props.item);
+	}
+
+	renderTreeViewItem(child, i) {
 		return (
 			<TreeViewItem
-				{...{
-					id: id && `${id}-${i}`,
-					item: child,
-					onSelect,
-					onClick,
-					depth: depth + 1,
-					key: i,
-				}}
+				id={`${this.props.id}-${i}`}
+				item={child}
+				onSelect={this.props.onSelect}
+				onClick={this.props.onClick}
+				depth={this.props.depth + 1}
+				key={i}
 			/>
 		);
 	}
 
-	function getIconAction(label, icon_, action, id_) {
+	renderIconAction(label, icon_, action, id_) {
 		return (
 			<Action
 				label={label}
 				icon={icon_}
-				onClick={getActionHandler(action, item)}
+				onClick={event => {
+					event.stopPropagation();
+					action(this.props.item);
+				}}
 				tooltipPlacement="right"
 				hideLabel
 				key={label}
-				id={id_ || `${id}-${icon_}`}
+				id={id_ || `${this.props.id}-${icon_}`}
```

Comment:
We should not have id if no `this.props.id`

Reply:
done in #880